cli/src/commands/init.rs

```diff
@@ -23,28 +27,52 @@ pub async fn handle_init(api: &mut PhylumApi, matches: &ArgMatches) -> CommandRe
         if !should_continue {
             return Ok(ExitCode::ProjectAlreadyInitialized.into());
         }
+
+        println!();
     }
 
     let cli_project = matches.get_one::<String>("project");
     let cli_group = matches.get_one::<String>("group");
 
     // Interactively prompt for missing information.
-    let (project, group) = prompt(cli_project, cli_group)?;
+    let (project, group) = prompt_project(cli_project, cli_group)?;
 
     // Attempt to create the project.
-    let response = project::create_project(api, &project, group.clone()).await?;
+    let result = project::create_project(api, &project, group.clone()).await;
 
     // If project already exists, just link to it.
-    match response {
-        CommandValue::Code(ExitCode::AlreadyExists) => {
-            project::link_project(api, &project, group).await
+    let mut project_config = match result {
+        Err(PhylumApiError::Response(ResponseError { code: StatusCode::CONFLICT, .. })) => {
+            print_user_success!("Successfully linked to project {project:?}");
+            project::link_project(api, &project, group).await.context("Unable to link project")?
         },
-        command_value => Ok(command_value),
-    }
+        project_config => {
+            if project_config.is_ok() {
+                print_user_success!("Successfully created project {project:?}");
+            }
+            project_config.context("Unable to create project")?
+        },
+    };
+
+    let cli_lockfile_type = matches.get_one::<String>("lockfile-type");
+    let cli_lockfile = matches.get_one::<String>("lockfile");
+
+    // Add lockfile and its type to the project configuration.
+    println!();
+    let (lockfile, lockfile_type) = prompt_lockfile(cli_lockfile, cli_lockfile_type)?;
+    project_config.lockfile_type = lockfile_type;
+    project_config.lockfile = Some(lockfile);
+
+    // Save project config.
+    config::save_config(Path::new(PROJ_CONF_FILE), &project_config).unwrap_or_else(|err| {
+        print_user_failure!("Failed to save project file: {}", err);
+    });
```

Comment:
It definitely seems like this should cause the entire command to fail rather than just printing the failure

```suggestion
    // Save project config.
    config::save_config(Path::new(PROJ_CONF_FILE), &project_config).context("Failed to save project file")?;
```